### PR TITLE
LPS-64753

### PR DESCRIPTION
**Pull request by Ayoub3r**

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path [hunk 1]
```diff
@@ -35,6 +35,21 @@
...
 	<td></td>
 </tr>
+<tr>
+	<td>APPLY_CUSTOM_SIZE</td>
+	<td>//button[@aria-label='Apply Custom Size']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>BUTTON_EXIT</td>
+	<td>//div[contains(@class,'simulation-panel')]//button[contains(@class,'btn btn-monospaced')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path:45]: do you really need a div here? what happens if the programmer changes it to a "p" ?

do you really need the classes ordered on that way, what happens if the programmer changes it?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path [hunk 2]
```diff
@@ -35,6 +35,21 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>PREVIEW_APPLIED_CHANGES</td>
+	<td>//div[contains(text(),'Applied') and contains(@class, 'autofit-section')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path:50]: do you really need a div here? what happens if the programmer changes it to a "p" ?

File: portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+		FormViewBuilder.validateFieldLabel(fieldLabel = "Idade");
+
+		FormFields.editText(
+			fieldName = "Idade",
+			fieldValue = 21);
+
+		FormPortlet.clickSubmit();
+
+		Form.submitAgain();
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro:138]: This is an Assert without any assert?

File: portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			fieldName = "Nom et Prénom",
+			fieldValue = "Ilyass Hossni");
+
+		FormViewBuilder.validateFieldLabel(fieldLabel = "Âge");
+
+		FormFields.editText(
+			fieldName = "Âge",
+			fieldValue = 22);
+
+		FormPortlet.clickSubmit();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro:115]: This is an Assert without any assert?

File: portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+			fieldName = "Full Name",
+			fieldValue = "Ayoub Er-rfife");
+
+		FormViewBuilder.validateFieldLabel(fieldLabel = "Age");
+
+		FormFields.editText(
+			fieldName = "Age",
+			fieldValue = 21);
+
+		FormPortlet.clickSubmit();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro:95]: This is an Assert without any assert?

**Review by manoelcyreno — CHANGES_REQUESTED** [review]
@Ayoub3r 
Please, take a look on thee code:
1. do the fixes that I mentioned.
2. "Given:", "When:", "Then:" is not the pattern that we use on the code.
3. try to move all the variables you are defining hardcoded on the macro, for the testcase, file, and the macro will work based on that parameter ...
4. Sometimes your class name is "CanEditBlog" , but what does it mean? Can you edit with success? failure?
5. Into the When, you can not use Json calls, fix it.

File: portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro (hunk 5 above)
Comment (Ayoub3r) [reply portal-web/test/functional/com/liferay/portalweb/macros/PoshiExercises.macro:95]: Hello @manoelcyreno , I've read every comment you've made and I'll be sure to tackle each one of them.

**Pull request closed without merging**